### PR TITLE
shred: fix `permissions_set_readonly_false` clippy error

### LINKED ISSUE #4487: shred.rs: error: call to `set_readonly` with argument false
```

error: call to `set_readonly` with argument `false`
   --> src/uu/shred/src/shred.rs:465:9
    |
465 |         perms.set_readonly(false);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: on Unix platforms this results in the file being world writable
    = help: you can set the desired permissions using `PermissionsExt`. For more information, see
            https://doc.rust-lang.org/std/os/unix/fs/trait.PermissionsExt.html
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#permissions_set_readonly_false
    = note: `-D clippy::permissions-set-readonly-false` implied by `-D warnings`
```

### PR DESCRIPTION
**Pull request by miles170**
Closes #4487

### CODE REVIEW

File: src/uu/shred/src/shred.rs [hunk 1]
```diff
@@ -462,6 +466,14 @@ fn wipe_file(
...
         let metadata = fs::metadata(path).map_err_context(String::new)?;
         let mut perms = metadata.permissions();
+        #[cfg(target_family = "unix")]
+        // NOTE: set_readonly(false) makes the file world-writable on Unix.
+        #[allow(clippy::unnecessary_cast)]
+        // NOTE: S_IWUSR type is u16 on macOS.
+        perms.set_mode(perms.mode() | (S_IWUSR as u32));
+        #[cfg(not(target_family = "unix"))]
+        #[allow(clippy::permissions_set_readonly_false)]
+        // FIXME: Clippy still throws error on non-UNIX platforms.
```
Comment (tertsdiepraam) [inline src/uu/shred/src/shred.rs:476]: Is there a tracking issue for this false positive?

File: src/uu/shred/src/shred.rs [hunk 2]
```diff
@@ -462,6 +466,14 @@ fn wipe_file(
     if force {
         let metadata = fs::metadata(path).map_err_context(String::new)?;
         let mut perms = metadata.permissions();
+        #[cfg(target_family = "unix")]
```
Comment (tertsdiepraam) [inline src/uu/shred/src/shred.rs:469]: Does `#[cfg(unix)]` work too?

**Comment by tertsdiepraam** [comment]
> I have only sets the write access for the owner on UNIX platform, not sure should it sets to group or others?

You can test it:

```
❯ ls -l test
---------- 1 terts terts 77824 10 mrt 11:29 test
❯ shred test
shred: test: failed to open for writing: Permission denied
❯ shred test -f
❯ ls -l test
--w------- 1 terts terts 77824 10 mrt 11:31 test
```

File: src/uu/shred/src/shred.rs (hunk 1 above)
Comment (miles170) [reply src/uu/shred/src/shred.rs:476]: I didn't find one.

File: src/uu/shred/src/shred.rs (hunk 2 above)
Comment (miles170) [reply src/uu/shred/src/shred.rs]: I'm not sure why `#[cfg(unix)]` works on Windows too.

Comment (tertsdiepraam) [reply src/uu/shred/src/shred.rs]: Are you sure? It really should be disabled on windows: https://doc.rust-lang.org/reference/conditional-compilation.html#unix-and-windows

File: src/uu/shred/src/shred.rs (hunk 1 above)
Comment (tertsdiepraam) [reply src/uu/shred/src/shred.rs]: Thanks, could you reference it in the code too?

**Pull request merged**